clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -708,14 +709,18 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
                 wakeupTrigger.maybeTriggerWakeup();
 
                 updateAssignmentMetadataIfNeeded(timer);
-                final Fetch<K, V> fetch = pollForFetches(timer);
-                if (!fetch.isEmpty()) {
-                    if (fetch.records().isEmpty()) {
-                        log.trace("Returning empty records from `poll()` "
+                if (isGenerationKnown()) {
```

Comment:
If this check is false... where do we block? It seems like we busy loop with 100% CPU, but maybe I'm wrong

Reply:
The busy loop should not be too bad since the group metadata update event is added to the queue when also the new assignment is set in the membership manager.